docs/docs/integrating-with-other-libraries.md

```diff
@@ -0,0 +1,270 @@
+
+### Replacing String Based Rendering with React
+
+A common pattern in older web applications is to describe chunks of the DOM as a string and insert it into the DOM like so: `$el.html(htmlString)`. These points in a codebase are perfect for introducing React. Just rewrite the string based rendering as a React component.
+
+So the following jQuery implementation...
+```js
+const htmlString = '<button id="jquery-btn">jQuery</button>';
+$el.html(htmlString);
+$('jquery-btn').click(() => window.alert('jQuery button'));
+```
+
+...could be rewritten using a React Component.
+```js
+function Button() {
+  return <button id='react-btn'>React</button>;
+}
+ReactDOM.render(
+  <Button />,
+  document.getElementById('react-container'),
+  () => {
+    $('#react-btn').click(() => window.alert('React button'));
+  }
+);
+```
+
+From here you could start moving more logic into the component and begin adopting more common React practices.
+
+```js
+function Button({ handleClick }) {
+  return <button onClick={handleClick}>React</button>;
+}
+ReactDOM.render(
+  <Button handleClick={() => window.alert('React button v2')} />,
+  document.getElementById('react-container')
+);
+```
+
+[Try it on CodePen.](http://codepen.io/wacii/pen/RpvYdj?editors=1010)
+
+### Embedding React in a Backbone View
+
+Backbone Views typically use HTML strings, or string producing template functions, to create the content for their associated DOM element. Like before, this process may be replaced with a React component.
+
+Each view will have an associated component, and when the view renders, `ReactDOM.render()` is used to render the component into the view's `el`.
+
+```js
+function Component({ text }) {
+  return <p>{text}</p>;
+}
+
+const ComponentView = Backbone.View.extend({
+  render() {
+    const text = this.model.get('text');
+    ReactDOM.render(<Component text={text} />, this.el);
+    return this;
+  },
+  remove() {
+    ReactDOM.unmountComponentAtNode(this.el);
+    Backbone.View.prototype.remove.call(this);
+  }
+});
+```
+
+State is maintained in the view as `this.model` and passed to the component as props.
+
+In addition to normal cleanup, event listeners registered through React as well as component state should be removed by calling `ReactDOM.unmountComponentAtNode()`. This is something React normally calls for us, but because we are controlling the application with Backbone, we must do this manually to avoid leaking memory.
+
+[Try it on CodePen.](http://codepen.io/wacii/pen/OWZJMQ?editors=0010)
+
+## Integrating with Model Layers
+
+React has little opinion on how data is stored or updated and so React components can easily incorporate the model layer from other frameworks. Models may be consumed as is or extracted using containers for better separation of concerns. Both approaches will be demonstrated using Backbone.
```

Comment:
Maybe worth adding that in general we recommend one directional data flow such as React state, Flux, or Redux, but other approaches work too.